extensions-contrib/opentelemetry-extensions/src/main/java/org/apache/druid/data/input/opentelemetry/protobuf/OpenTelemetryMetricsProtobufReader.java

```diff
@@ -143,25 +149,38 @@ private InputRow parseNumberDataPoint(NumberDataPoint dataPoint,
 
     if (dataPoint.hasAsInt()) {
       event.put(valueDimension, dataPoint.getAsInt());
-    } else if (dataPoint.hasAsDouble()) {
-      event.put(valueDimension, dataPoint.getAsDouble());
     } else {
-      throw new IllegalStateException("Unexpected dataPoint value type. Expected Int or Double");
+      event.put(valueDimension, dataPoint.getAsDouble());
     }
 
     event.putAll(resourceAttributes);
     dataPoint.getAttributesList().forEach(att -> event.put(metricAttributePrefix + att.getKey(),
-        getStringValue(att.getValue())));
+        parseAnyValue(att.getValue())));
 
     return createRow(TimeUnit.NANOSECONDS.toMillis(dataPoint.getTimeUnixNano()), event);
   }
 
-  private static String getStringValue(AnyValue value)
+  private static Object parseAnyValue(AnyValue value)
   {
-    if (value.getValueCase() == AnyValue.ValueCase.STRING_VALUE) {
-      return value.getStringValue();
+    switch (value.getValueCase()) {
+      case INT_VALUE:
+        return value.getIntValue();
+      case BOOL_VALUE:
+        return value.getBoolValue();
+      case ARRAY_VALUE:
+        return value.getArrayValue();
+      case BYTES_VALUE:
+        return value.getBytesValue();
+      case DOUBLE_VALUE:
+        return value.getDoubleValue();
+      case KVLIST_VALUE:
+        return value.getKvlistValue();
+      case STRING_VALUE:
+        return value.getStringValue();
+      default:
+        // VALUE_NOT_SET:
+        return "";
```

Comment:
how does empty string get treated if someone uses a numeric column?

Reply:
It will return 0 as a default value of numeric column. But actually in the code, it would throw 
```
 throw new NumberFormatException("empty String");
```
and then get caught [here](https://github.com/confluentinc/druid/blob/1a09b2c847c8bb18694af2978e3bdb7f2496148e/core/src/main/java/org/apache/druid/data/input/Rows.java#L114) and return default value. 


Probably it's still better to return `null` here since druid would replace with default value according to different column specs directly [here](https://github.com/confluentinc/druid/blob/1a09b2c847c8bb18694af2978e3bdb7f2496148e/core/src/main/java/org/apache/druid/data/input/Rows.java#L101) instead of going throw the whole parsing process, even though the eventual result is the same.